src/components/main-menu/MainMenu.tsx

```diff
@@ -35,6 +38,17 @@ const MainMenu = Object.assign(
 
       return (
         <MainMenuTunnel.In>
+          {appState.openMenu === "canvas" && device.isMobile && (
+            <Portal.Root
```

Comment:
This is similar to `onClickOutside` ? Can you elaborate on why is this needed specific for mobile?

Reply:
Sure.

Please refer to https://github.com/excalidraw/excalidraw/pull/6483#issuecomment-1521566179 for the gist of it and some of the preceding discussion for context. The TL;DR is that we want to allow submenu nesting on mobile too and as such we don't want a full-width menu on mobile. We also don't want to allow clicks on canvas while the mobile menu is open, as it can be especially aggravating when the welcome screen is shown and you "fat finger click" behind the menu and open something you never intended to. So the solution we agreed on with @dwelle was that:
- on mobile, stick with a non full width main menu to allow nesting/submenus
- on mobile, show a backdrop while the main menu is open; clicking on the backdrop should close the menu
- on desktop, no such restriction is needed

Reply:
Sounds good 👍🏻